spring-integration-http/src/main/java/org/springframework/integration/http/outbound/HttpRequestExecutingMessageHandler.java

```diff
@@ -360,8 +360,10 @@ protected void doInit() {
 			ConverterRegistry converterRegistry =
 					(ConverterRegistry) conversionService;
 
-			converterRegistry.addConverter(new ClassToStringConverter());
-			converterRegistry.addConverter(new ObjectToStringConverter());
+			if (!conversionService.canConvert(Class.class, String.class)) {
```

Comment:
Gary, I found that these converters (and therefore memory leak fix) is relevant only for the 3.0.x.
In 4.0 I introduced this one: https://jira.spring.io/browse/INT-3052, hence we can simply get deal with `Class<?>` result from the SpEL evaluation without conversion to String.And the removal of these converters is feasible even for the 4.0.x.